cmd/loadgenerator/main.go

```diff
@@ -0,0 +1,222 @@
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "unable to check status : %v", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		if !isSpendRound(cfg, nodeStatus.LastRound) {
+			return
+		}
+	waitForNextBlock:
+		// wait for the next round.
+		nodeStatus, err = restClient.WaitForBlock(basics.Round(nodeStatus.LastRound))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "unable to wait for next round node status : %v", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		if !isSpendRound(cfg, nodeStatus.LastRound) {
+			return
+		}
+		goto waitForNextBlock
+	}
+}
+
+func waitForSpendingRound(restClient client.RestClient, cfg config) {
+	var nodeStatus generatedV2.NodeStatusResponse
+	var err error
+	for {
+		nodeStatus, err = restClient.Status()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "unable to check status : %v", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		if isSpendRound(cfg, nodeStatus.LastRound) {
+			// time to send transactions.
+			return
+		}
+		fmt.Printf("Current round %d, waiting for spending round %d\n", nodeStatus.LastRound, nextSpendRound(cfg, nodeStatus.LastRound))
+	waitForNextBlock:
+		// wait for the next round.
+		nodeStatus, err = restClient.WaitForBlock(basics.Round(nodeStatus.LastRound))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "unable to wait for next round node status : %v", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		if isSpendRound(cfg, nodeStatus.LastRound) {
+			// time to send transactions.
+			return
+		}
+		goto waitForNextBlock
+	}
+}
+
+func generateTransactions(restClient client.RestClient, cfg config, privateKey *crypto.SignatureSecrets, publicKey basics.Address) (queueFull bool) {
+	var nodeStatus generatedV2.NodeStatusResponse
+	var err error
+	nodeStatus, err = restClient.Status()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to check status : %v", err)
+		return false
+	}
+	var vers common.Version
+	vers, err = restClient.Versions()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to get versions : %v", err)
+		return false
+	}
+	var genesisHash crypto.Digest
+	copy(genesisHash[:], vers.GenesisHash)
+	// create transactionBlockSize transaction to send.
+	txns := make([]transactions.SignedTxn, transactionBlockSize, transactionBlockSize)
+	for i := range txns {
+		tx := transactions.Transaction{
+			Header: transactions.Header{
+				Sender:      publicKey,
+				Fee:         basics.MicroAlgos{Raw: cfg.Fee},
+				FirstValid:  basics.Round(nodeStatus.LastRound),
+				LastValid:   basics.Round(nodeStatus.LastRound + 2),
+				Note:        make([]byte, 4),
+				GenesisID:   vers.GenesisID,
+				GenesisHash: genesisHash,
+			},
+			PaymentTxnFields: transactions.PaymentTxnFields{
+				Receiver: publicKey,
+				Amount:   basics.MicroAlgos{Raw: 0},
+			},
+			Type: protocol.PaymentTx,
+		}
+		crypto.RandBytes(tx.Note[:])
+		txns[i] = tx.Sign(privateKey)
+	}
+
+	// create multiple go-routines to send all these requests.
+	const nroutines = 8
```

Comment:
Maybe this should be parametrized either in the config or as a constant at the top?

Reply:
I'll make it equal to twice the number of cpus.